app/models/chef.rb

```diff
@@ -1,4 +1,10 @@
 class Chef < ApplicationRecord
    validates_presence_of :name
    has_many :dishes
+   has_many :dish_ingredients, through: :dishes
+   has_many :ingredients, through: :dish_ingredients
+
+   def list_ingredients
+      self.ingredients.distinct
```

Comment:
Excellent work! Adding the additional association definitely makes this logic much easier.